lib/compilers/dex2oat.ts

```diff
@@ -89,7 +87,7 @@ export class Dex2OatCompiler extends BaseCompiler {
         this.methodRegex = /^\s+\d+:\s+(.*)\s+\(dex_method_idx=\d+\)$/;
         this.methodSizeRegex = /^\s+CODE:\s+\(code_offset=0x\w+\s+size=(\d+).*$/;
         this.insnRegex = /^\s+(0x\w+):\s+\w+\s+(.*)$/;
-        this.stackMapRegex = /^\s+(StackMap\[\d+\])\s+\((.*)\).*$/;
+        this.stackMapRegex = /^\s+(StackMap\[\d+])\s+\((.*)\).*$/;
```

Comment:
Really not sure about this. I think the backslash is needed.

Reply:
I think because there's no starting `[`, the regex parser won't look for a closing `]`, so you don't need to escape it.

It's super unclear of course, because someone else might be looking at this code and be like, "hey the first `[` shouldn't be escaped" I better remove it, and then it completely breaks

Reply:
![Screenshot from 2024-03-31 21-33-45](https://github.com/compiler-explorer/compiler-explorer/assets/442630/b6ccdc54-d0a6-49fc-a408-8cf58a78df29)

so still matches correctly

not sure how to disable this specific one

Reply:
Definitely confused me. Maybe better to disable `unicorn/better-regex` for this line just for the sake of readability.

Reply:
ah, good idea!